modules/vr_editor/vr_collision.h

```diff
@@ -0,0 +1,86 @@
+/*************************************************************************/
+/*  vr_collision.h                                                       */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef VR_COLLISION_H
+#define VR_COLLISION_H
+
+#include "scene/3d/node_3d.h"
+
+// VRCollision is a helper class for our poke and grab detection nodes to test if we're interacting with a shape.
+// We can't use the physics engine in the editor so this is a bare minimum implementation suitable specifically
```

Comment:
What's the limitation with regard to using the physics engine in the editor? Can we bypass it only for the VR Editor?

Reply:
This worried me because physics is such a necessary thing when making XR tool scripts. Luckily, a user can turn on Physics like this:
![image](https://user-images.githubusercontent.com/30942833/210271517-a067770f-609b-4368-9cf5-a4fa1c780076.png)

But if your Tools use physics (like a XR Editor Locomotion Tool) is there a way enable physics only on specific Nodes?

Enabling physics like this made all my RigidBody3D nodes start to move in my Editor scene even though they don't have a Tool script active. It seems like, like with @tool and GdScript, we would need a new user friendly way to toggle physics On and Off on a Node-specific basis. So XR tools can use physics without everything using physics.

We have the Freeze toggle: 
![image](https://user-images.githubusercontent.com/30942833/210272016-1c882d82-cdc3-4f85-bd0e-dd0b723a2c4a.png)

But then you have to make a Tool script that searches through your entire Scene tree and Freezes everything while editing and Unfreezes everything when you run the project in non-editor mode.

Reply:
This needs further discussion with the core team as there must be a good reason the physics engine is disabled in editor and enabling it would cause those issues.

That said, I was not originally aware the physics engine could be enabled hence building a simpler version of the features needed for XR interactions. It would be way easier if we can just rely on the physics engine.